src/kinetics/Reaction.cpp

```diff
@@ -211,6 +222,126 @@ void Reaction::calculateRateCoeffUnits(const Kinetics& kin)
     }
 }
 
+void updateUndeclared(std::vector<std::string>& undeclared,
+                      const Composition& comp, const Kinetics& kin) {
+    for (const auto& sp: comp) {
+        if (kin.kineticsSpeciesIndex(sp.first) == npos) {
+            undeclared.emplace_back(sp.first);
+        }
+    }
+}
+
+std::pair<std::vector<std::string>, bool> Reaction::undeclaredThirdBodies(
+        const Kinetics& kin) const {
+    std::vector<std::string> undeclared;
+    return std::make_pair(undeclared, false);
+}
+
+void Reaction::checkBalance(const Kinetics& kin) const {
+
+    Composition balr, balp;
```

Comment:
If you want the visual space here, I'd suggest
```suggestion
void Reaction::checkBalance(const Kinetics& kin) const
{
    Composition balr, balp;
```
Cantera's code isn't particularly consistent about this minor formatting detail, so I think it's fine with the brace in either place, but that might be a way of solving our opposite preferences about blank lines at the start of a block. The one case where I do think having the brace on a new line by itself is especially useful is when the function signature end up being split across multiple lines so you don't end up with a weird looking dedent.

Reply:
Not a problem. I don't have hard preferences when it comes to coding style.